Program.cs

```diff
@@ -13,23 +15,141 @@ private static bool isDivisibleBy(int number, int divisior)
-            Console.WriteLine("Buzz");
-        }
-        else if (isDivisibleBy(number, 3))
+            { 3, CompositionActions.AppendFizz },
+            { 5, CompositionActions.AppendBuzz },
+            { 7, CompositionActions.AppendBang },
+            { 11, CompositionActions.ReplaceEverythingWithBong },
+            { 13, CompositionActions.SuffixWithFezz },
+            { 17, CompositionActions.ReverseAll }
+        };
+
+        var answerComponents = new List<CompositionActions> { };
+        bool showNumber = true;
+
+        foreach (var divisor in divisorsAndNames.Keys)
+            if (isDivisibleBy(number, divisor))
+            {
+                var action = divisorsAndNames[divisor];
+                answerComponents.Add(divisorsAndNames[divisor]);
+                if (action != CompositionActions.ReverseAll)
+                    showNumber = false;
+            }
+
+        if (showNumber)
+            return number.ToString();
+        else
+            return composeString(answerComponents.ToArray());
+    }
+
+    private static string composeString(CompositionActions[] list)
+    {
+        var output = new List<string>();
+        int indexOfFirstB = -1;
+
+        foreach (var action in list)
         {
-            Console.WriteLine("Fizz");
+            switch (action)
+            {
+                case CompositionActions.AppendFizz:
+                    {
+                        output.Add("Fizz");
+                        break;
+                    }
+                case CompositionActions.AppendBuzz:
+                case CompositionActions.AppendBang:
+                    {
+                        string name = action.ToString().Replace("Append", "");
+
+                        if (indexOfFirstB < 0)
+                        {
+                            indexOfFirstB = output.Count;
+                        }
+
+                        output.Add(name);
+                        break;
+                    }
+                case CompositionActions.ReplaceEverythingWithBong:
+                    {
+                        output = new List<string> { "Bong" };
+                        indexOfFirstB = 0;
+                        break;
+                    }
+                case CompositionActions.SuffixWithFezz:
+                    {
+                        int insertFezzAtIndex = output.Count;
+
+                        if (indexOfFirstB >= 0)
+                        {
+                            insertFezzAtIndex = indexOfFirstB;
+                        }
+
+                        output.Insert(insertFezzAtIndex, "Fezz");
+                        break;
+                    }
+                case CompositionActions.ReverseAll:
+                    {
+                        output.Reverse();
+                        break;
+                    }
+            }
         }
-        else
+        return String.Join("", output.ToArray());
+    }
+
+    public static void testResponseToNumber()

```

Comment:
Great work adding tests! 

Usually we don't store the tests in the same place as the main program (in this case its just an exercise so its fine) but if you find yourself out of things to do then a fun stretch goal could be to:
- Set up the test files separately and add instructions on how to run them
- Set up "proper" unit testing using NUnit https://docs.nunit.org/articles/nunit/getting-started/installation.html